rebasehelper/utils.py

```diff
@@ -202,22 +202,29 @@ def capture_output(func, capture_stdout=False, capture_stderr=False):
         return stdout_data, stderr_data
 
 
+class DownloadError(Exception):
+    """Exception indicating that download of a file failed."""
+    pass
+
+
 class DownloadHelper(object):
 
     """Class for downloading sources defined in SPEC file"""
 
     @staticmethod
-    def download_file(url, destination_name):
+    def download_file(url, destination_path, timeout=10, blocksize=8192):
         """
-        Method for downloading file using pycurl
+        Method for downloading file from HTTP, HTTPS and FTP URL.
 
         :param url: URL from which to download the file
-        :param destination_name: path where to store downloaded file
+        :param destination_path: path where to store downloaded file
+        :param timeout: timeout in seconds for blocking actions like connecting, etc.
+        :param blocksize: size in Bytes of blocks used for downloading the file and reporting progress
         :return: None
         """
-        def progress(download_total, downloaded, upload_total, uploaded):
+        def progress(download_total, downloaded, start_time):
             r = downloaded / download_total if download_total else 0.0
```

Comment:
Sorry, I forgot to mention the problem is with python2 only. With python3 it works as expected.